avro-serializer/src/test/java/io/confluent/kafka/serializers/KafkaAvroSerializerTest.java

```diff
@@ -162,6 +195,23 @@ public void testKafkaAvroSerializerSpecificRecord() {
   }
 
   @Test
+  public void testKafkaAvroSerializerSpecificRecordWithProjection() {
+    byte[] bytes;
+    Object obj;
+
+    IndexedRecord avroRecord = createExtendedSpecificAvroRecord();
+    bytes = avroSerializer.serialize(topic, avroRecord);
+
+    obj = specificAvroDecoder.fromBytes(bytes);
+    assertTrue("Full object should be a io.confluent.kafka.example.ExtendedUser", ExtendedUser.class.isInstance(obj));
+    assertEquals(avroRecord, obj);
+
+    obj = specificAvroDecoder.fromBytes(bytes, User.getClassSchema());
+    assertTrue("Projection object should be a io.confluent.kafka.example.User", User.class.isInstance(obj));
+    assertEquals("testUser", ((User)obj).getName().toString());
+  }
```

Comment:
I didn't realize that there was one.  I wrote the first version of this against an earlier confluent version and didn't see that.